server/src/main/java/com/hedera/block/server/pbj/PbjBlockAccessServiceProxy.java

```diff
@@ -0,0 +1,157 @@
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.block.server.pbj;
+
+import static com.hedera.block.server.metrics.BlockNodeMetricTypes.Counter.SingleBlocksNotFound;
+import static com.hedera.block.server.metrics.BlockNodeMetricTypes.Counter.SingleBlocksRetrieved;
+import static java.lang.System.Logger.Level.DEBUG;
+import static java.lang.System.Logger.Level.ERROR;
+
+import com.hedera.block.server.config.BlockNodeContext;
+import com.hedera.block.server.metrics.MetricsService;
+import com.hedera.block.server.persistence.storage.read.BlockReader;
+import com.hedera.block.server.service.ServiceStatus;
+import com.hedera.hapi.block.SingleBlockRequest;
+import com.hedera.hapi.block.SingleBlockResponse;
+import com.hedera.hapi.block.SingleBlockResponseCode;
+import com.hedera.hapi.block.stream.Block;
+import com.hedera.pbj.runtime.ParseException;
+import com.hedera.pbj.runtime.grpc.Pipeline;
+import com.hedera.pbj.runtime.grpc.Pipelines;
+import com.hedera.pbj.runtime.io.buffer.Bytes;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.io.IOException;
+import java.util.Optional;
+import java.util.concurrent.Flow;
+import javax.inject.Inject;
+
+/**
+ * PbjBlockAccessServiceProxy is the runtime binding between the PBJ Helidon Plugin and the
+ * Block Node. The Helidon Plugin routes inbound requests to this class based on the methods
+ * and service names in PbjBlockAccessService. Service implementations are instantiated via
+ * the open method thereby bridging the client requests into the Block Node application.
+ */
+public class PbjBlockAccessServiceProxy implements PbjBlockAccessService {
+
+    private final System.Logger LOGGER = System.getLogger(getClass().getName());
+
+    private final ServiceStatus serviceStatus;
+    private final BlockReader<Block> blockReader;
+    private final MetricsService metricsService;
+
+    /**
+     * Creates a new PbjBlockAccessServiceProxy instance.
+     *
+     * @param serviceStatus the service status
+     * @param blockReader the block reader
+     * @param blockNodeContext the block node context
+     */
+    @Inject
+    public PbjBlockAccessServiceProxy(
+            @NonNull final ServiceStatus serviceStatus,
+            @NonNull final BlockReader<Block> blockReader,
+            @NonNull final BlockNodeContext blockNodeContext) {
+        this.serviceStatus = serviceStatus;
+        this.blockReader = blockReader;
+        this.metricsService = blockNodeContext.metricsService();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    @NonNull
```

Comment:
are we still using @NonNull annotation on this cases? I thought we are only using it on parameters, constructors and such.

Reply:
I thought the convention was to annotate with `@NonNull` if there was a non-primative return type. I see that throughout the codebase.